torch/_refs/__init__.py

```diff
@@ -2748,8 +2748,16 @@ def native_layer_norm(
 # TODO: Adding this as a meta function causes functorch tests to fail when compiled with debug mode.
 # test/test_eager_transforms.py::TestFunctionalizeCPU::test_functionalize_fx_transpose_simple_cpu
 @register_decomposition(torch.ops.aten.permute, disable_meta=True)
-def permute(a: TensorLikeType, dims: DimsSequenceType) -> TensorLikeType:
-    _permutation = utils.canonicalize_dims(a.ndim, dims)
+def permute(a: TensorLikeType, *dims: Union[DimsSequenceType, Tuple[DimsSequenceType, ...]]) -> TensorLikeType:
+    dims_: DimsSequenceType
+    if dims and isinstance(dims[0], Sequence):
+        assert len(dims) == 1
+        dims = cast(Tuple[DimsSequenceType], dims)
+        dims_ = dims[0]
+    else:
+        dims_ = cast(DimsSequenceType, dims)
+
+    _permutation = utils.canonicalize_dims(a.ndim, dims_)
```

Comment:
Here's a utility function used for the reshape and view:
```suggestion
    _permutation = utils.extract_shape_from_varargs(dims, validate=False)
```

Reply:
The thing that had me hesitate was that it's dimensions, not shape here. Of course, moving it to a utility function could work regardless.

Reply:
If you had opinions on dims vs. sizes, I'd be glad to hear them.
To my mind, the options are:
- keep as is,
- have a dims helper similar to the shape one,
- use the shape one.

But I'm torn which is the best, I'm leaning towards the top half of the list.

Reply:
Let's move this to a dims-specific helper function now as it's likely to be useful in other places.

Reply:
Subito! Thank you for your input @jjsjann123 and @IvanYashchuk .